pkg/artifactory/resource_artifactory_group.go

```diff
@@ -55,74 +53,97 @@ func resourceArtifactoryGroup() *schema.Resource {
+	}
+
+	if realmAttributes := d.getStringRef("realm_attributes", false); realmAttributes != nil {
+		group.RealmAttributes = *realmAttributes
+	}
 
-	group.Name = d.getStringRef("name", false)
-	group.Description = d.getStringRef("description", false)
-	group.AutoJoin = d.getBoolRef("auto_join", false)
-	group.AdminPrivileges = d.getBoolRef("admin_privileges", false)
-	group.Realm = d.getStringRef("realm", false)
-	group.RealmAttributes = d.getStringRef("realm_attributes", false)
+	if usersNames := d.getSetRef("users_names"); usersNames != nil {
+		group.UsersNames = *usersNames
+	}
 
 	// Validator
-	if group.AdminPrivileges != nil && group.AutoJoin != nil && *group.AdminPrivileges && *group.AutoJoin {
-		return nil, fmt.Errorf("error: auto_join cannot be true if admin_privileges is true")
+	if group.AdminPrivileges && group.AutoJoin {
+		return artifactory.GroupParams{}, fmt.Errorf("error: auto_join cannot be true if admin_privileges is true")
 	}
 
-	return group, nil
+	return artifactory.GroupParams{GroupDetails: group, ReplaceIfExists: true, IncludeUsers: true}, nil
 }
 
 func resourceGroupCreate(d *schema.ResourceData, m interface{}) error {
-	c := m.(*ArtClient).ArtOld
-
-	group, err := unmarshalGroup(d)
+	c := m.(*ArtClient).ArtNew
 
+	groupParams, err := groupParams(d)
 	if err != nil {
 		return err
 	}
 
-	_, err = c.V1.Security.CreateOrReplaceGroup(context.Background(), *group.Name, group)
-
+	err = c.CreateGroup(groupParams)
 	if err != nil {
 		return err
 	}
 
-	d.SetId(*group.Name)
+	d.SetId(groupParams.GroupDetails.Name)
 	return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
-		c := m.(*ArtClient).ArtOld
-		_, resp, err := c.V1.Security.GetGroup(context.Background(), d.Id())
+		exists, err := resourceGroupExists(d, m)
 		if err != nil {
 			return resource.NonRetryableError(fmt.Errorf("error describing group: %s", err))
 		}
 
-		if resp.StatusCode == http.StatusNotFound {
+		if !exists {
 			return resource.RetryableError(fmt.Errorf("expected group to be created, but currently not found"))
 		}
 
 		return resource.NonRetryableError(resourceGroupRead(d, m))
 	})
 }
 
+func resourceGroupGet(d *schema.ResourceData, m interface{}) (*artifactory.Group, error) {
+	c := m.(*ArtClient).ArtNew
+
+	params := artifactory.NewGroupParams()
+	params.GroupDetails.Name = d.Id()
+	params.IncludeUsers = true
+
+	return c.GetGroup(params)
+}
+
 func resourceGroupRead(d *schema.ResourceData, m interface{}) error {
-	c := m.(*ArtClient).ArtOld
-
-	group, resp, err := c.V1.Security.GetGroup(context.Background(), d.Id())
-	if resp == nil {
-		return fmt.Errorf("no response returned during resourceGroupRead")
-	}
-	// If we 404 it is likely the resources was externally deleted
-	// If the ID is updated to blank, this tells Terraform the resource no longer exist
-	if resp.StatusCode == http.StatusNotFound {
-		d.SetId("")
-		return nil
-	} else if err != nil {
+	group, err := resourceGroupGet(d, m)
```

Comment:
I don't know why the original status code check was there, but I am not comfortable removing it without an explanation